### DISCUSSION TITLE
Overview of Features: Kerberos.io solution stack

### DISCUSSION THREAD

**Discussion by cedricve** [kerberos-io/agent#140] [General]
This discussion includes all features included in the Kerberos.io solution stack grouped by solution:

# Kerberos Agent / Kerberos Factory

- Low memory and CPU usage.
- Simplified and modern user interface.
- Multi architecture (ARMv6, ARMv7, ARM64, AMD64)
- Multi stream, for example recording in H265, live streaming and motion detection in H264.
- Multi camera support: IP Cameras (H264 and H265), USB cameras and Raspberry Pi Cameras [through a RTSP proxy](https://github.com/kerberos-io/camera-to-rtsp).
- Single camera per instance (e.g. one container per camera).
- Low resolution streaming through MQTT and high resolution streaming through WebRTC (only supports H264/PCM).
- Backchannel audio from Kerberos Hub to IP camera (requires PCM ULAW codec)
- Audio (AAC) and video (H264/H265) recording in MP4 container.
- End-to-end encryption through MQTT using RSA and AES (livestreaming, ONVIF, remote configuration, etc)
- Conditional recording: offline mode, motion region, time table, continuous recording, webhook condition etc.
- Post- and pre-recording for motion detection.
- Encryption at rest using AES-256-CBC.
- Ability to create fragmented recordings, and streaming through HLS fMP4.
- [Deploy where you want](#how-to-run-and-deploy-a-kerberos-agent) with the tools you use: `docker`, `docker compose`, `ansible`, `terraform`, `kubernetes`, etc.
- Cloud storage/persistance: Kerberos Hub, Kerberos Vault and Dropbox. [(WIP: Minio, Storj, Google Drive, FTP etc.)](https://github.com/kerberos-io/agent/issues/95)
- Outputs: trigger an integration (Webhooks, MQTT, Script, etc) when a specific event (motion detection or start recording ) occurs
- REST API access and documentation through Swagger (trigger recording, update configuration, etc).
- MIT License

# Kerberos Vault

- Bring your own storage: Google Cloud Storage, Azure Storage, S3 Compliant storage: AWS S3, Ceph, Minio, Wasabi, Storj, etc.
- Hybrid forwarding and chaining: Kerberos Vault can be configured in a hybrid scenario where one Kerberos Vault is deployed at the edge and buffering recordings, while another Kerberos Vault is deployed in the cloud. Depending on the scenario Kerberos Vault can synchronise to a remote Kerberos Vault depending on the scenario (Machine learning, time-based, or customer scenario.
- Custom integrations through message brokers: RabbitMQ (or other AMQP broker), SQS and KAfka
- Deployment through Docker, Docker compose or Kubernetes.
- REST API access and documentation through Swagger (read recordings, update configuration, etc).
- Machine learning: Ability to integrate machine learning models through the integration functionality. Data is stored and made available to data scientist through the concept of a message broker. Models can run concurrently to process high loads of data.

# Kerberos Hub

- Deployed on top Kubernetes through Helm charts.
- Live viewing: Ability to live stream low resolution (MJPEG) and or high resolution (WebRTC)
- Camera governance: ability to organise cameras in groups or sites.
- Fine grained user access control: various users can get access to the platform using SSO or username-password, and obtain access to specific sites, groups and/or cameras under a specific role (gues, admin or custom).
- Video walls: ability to create (public/private) video walls so external users or devices can have access to live views without the need of providing full access.
- Alerts: Configuring specific events such as object crossed line, motion was detected, camera went offline, and more. 
- REST API access and documentation through Swagger (retrieve data such as users, cameras, sites, etc, update configuration, and much more).

**Comment by mpgreg** [comment]
@cedricve  you mention REST API access and swagger docs.  Is that implemented?

**Reply by KilianBoute** [reply]
Hey, yes it is. You can find the swagger docs here: https://api.cloud.kerberos.io/swagger/index.html

**Reply by cedricve** [reply]
I think he is looking for that Agent swagger?
-> https://demo.kerberos.io/swagger/index.html
